Sources/Mimus/Mock.swift

```diff
@@ -81,21 +81,24 @@ public extension Mock {
         let callCandidates = storage.recordedCalls.filter {
             $0.identifier == callIdentifier
         }
-        
-        // Note we explicitly reset storage _before_ triggering verification as verification might add new values to
-        // the storage which we want to keep around for additional verification (e.g. when simulating callbacks).
-        storage.reset()
+
+        // Note we explicitly remove all calls of a given identified from the storage _before_ triggering verification
+        // ... as verification might add new values to the storage which we want to keep around for additional verification
+        // (e.g. when simulating callbacks).
+        storage.remove(callsWithIdentifier: callIdentifier)
```

Comment:
I think we shouldn't remove _all_ recorded calls with this identifier, AFAIK other frameworks (notably OCMockito) ignores callbacks that match both signature _and_ parameters.

Reply:
👍 
Had to rebuild how the arguments matching is done, but, in the end, only records that are an exact match are removed from storage.
🙏 Take a look.